assets/js/consent-mode/consent-mode.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+( function () {
+	document.addEventListener(
+		'wp_listen_for_consent_change',
+		function ( event ) {
```

Comment:
How about we extract this to a named function as well (similar to `updateGrantedConsent` below) to avoid extra indentation and wrapping from longer lines?